clients/csharp-dotnet-core/voice-assistant-test/README.md

```diff
@@ -174,14 +174,41 @@ Here is the full list:
+>>##### ExpectedTTSAudioResponseDurations 
+>>`array of strings | optional | null | ["1500", "-1", "2000"]`. Expected duration (in msec) of bot response TTS audio stream. This allows you to validate that the right audio stream duration was downloaded by the tool. Otherwise the test will fail. The length of this array (if exists) must match the length of the [ExpectedResponses](#expectedresponses) array. Not every bot-response will have a TTS audio stream associated with it. In that case, specify a value of -1 in the array cell. The expected duration does not have to exactly match the actual duration for the test to succeed. This is controlled by the field [TTSAudioDurationMargin](#ttsaudiodurationmargin). Only if the different between expected duration and actual duration is outside this margin, the test will fail. See the [Getting Started Guide](docs/GETTING-STARTED-GUIDE.md) for an example of using ExpectedTTSAudioResponseDurations.
 >>
 >>##### ExpectedResponseLatency 
 >>`string | optional | null | "500", or "500,0" or "500,1"'. The expected time the tool should have received a particular bot-response activity. If the tool did not receive that activity by this latency value, the test will fail. There are two formats for the string. The first one just includes a positive integer. In this case the bot-response that is timed is the last one expected to arrive (based on the length of the [ExpectedResponses](#ExpectedResponses) array]). So for example of the length of ExpectedResponses is 3, it means the tool will wait until it receives three bot response activities. If the 3rd one was not received by the time specified by  Expectedresponse, the test will fail. The second format of the string is a positive integer (the duration), followed by a comma, followed by a zero-based index. The index specifies which of the bot-response activities should be time-measured, which 0 being the first one specified in the ExpectedResponses array. This second format allows you to put an upper limit on either one of the bot responses, not just the last one.
 
 ## Topics
 
+### Testing response activities with random text and speak from a predefined set
+
+In some cases, Bot or Custom Commands application can define a set of text or speech for responding, and randomly select to use one of them in actual responded activity. For example, a command of Custom Commands application defines 2 sentences asking for the temperature to change - 1. How much do you want to change the temperature by? 2. By how many degrees? This command will randomly use the first sentence in some responses and the second sentence in other responses. To deal with this case, you can use " || " in text, speak of ExpectedResponses field along with ExpectedTTSAudioResponseDurations field to specify the expected responses as below:

```

Comment:
... temperature to change: (1) How much ... (2) by how ...

Reply:
Done.